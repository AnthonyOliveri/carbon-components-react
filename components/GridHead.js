import GridHeadCell from './GridHeadCell';
import React from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';

export default class GridHead extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      resizeColumn: null,
      columns: props.columns,
      leftMarkerPosition: -9999,
      rightMarkerPosition: -9999,
      markerPositionTop: -9999,
    };
    this.calculateNewWidth = this.calculateNewWidth.bind(this);
    this.dragEnd = this.dragEnd.bind(this);
    this.dragStart = this.dragStart.bind(this);
    this.dragOver = this.dragOver.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      columns: nextProps.columns,
    });
  }

  calculateNewWidth(e, updatedColumn) {
    const newColumns = [];
    let cumulativePercentage = 0;
    let percentageChange = 0;
    this.state.columns.forEach(column => {
      const newColumn = Object.assign({}, column);
      if (!percentageChange) {
        cumulativePercentage += column.width;

        if (column.title === updatedColumn.title) {
          const currentWidthInPx = ReactDOM.findDOMNode(
            this.cells[updatedColumn.title]
          ).getBoundingClientRect().width;
          percentageChange =
            column.width *
            (e.clientX - this.state.resizeColumn.dragStartLocation) /
            currentWidthInPx;
          const newWidth = column.width + percentageChange;

          // Ensure the minimum size of the column to 5% while reducing the size
          if (newWidth < 5) {
            percentageChange = 5 - column.width;
            newColumn.width = 5;
          } else {
            newColumn.width = newWidth;
          }
        }
      } else {
        // Set remaining column widths proportionally
        newColumn.width =
          column.width *
          (100 - percentageChange - cumulativePercentage) /
          (100 - cumulativePercentage);
      }
      newColumns.push(newColumn);
    });

    this.setState({
      columns: newColumns,
    });
    return newColumns;
  }

  dragEnd(e, column) {
    const newColumns = this.calculateNewWidth(e, column);
    this.setState({
      columns: newColumns,
      resizeColumn: null,
      leftMarkerPosition: -9999,
      rightMarkerPosition: -9999,
      markerPositionTop: -9999,
    });
    this.props.updateColumns(newColumns);
  }

  dragOver(e) {
    const gridWidth = ReactDOM.findDOMNode(this).getBoundingClientRect().width;

    if (e.clientX - this.state.leftMarkerPosition > gridWidth * 0.05) {
      this.setState({
        rightMarkerPosition: e.clientX,
      });
    }
  }

  dragStart(e, column) {
    const gridDimensions = ReactDOM.findDOMNode(
      this.cells[column.title]
    ).getBoundingClientRect();
    this.setState({
      resizeColumn: {
        column,
        dragStartLocation: e.clientX,
      },
      leftMarkerPosition: gridDimensions.left,
      rightMarkerPosition: e.clientX,
      markerPositionTop: gridDimensions.top,
    });
  }

  render() {
    const cells = this.state.columns.map(column => (
      <GridHeadCell
        className="bx--grid-head__cell"
        column={column}
        dragEnd={this.dragEnd}
        dragOver={this.dragOver}
        dragStart={this.dragStart}
        key={column.title}
        onSort={this.props.onSort}
        ref={cell => {
          if (!this.cells) {
            this.cells = [];
          }
          if (cell) {
            this.cells[column.title] = cell;
          }
        }}
        sort={this.props.sort}
      />
    ));

    return (
      <div className="bx--grid-head">
        <div
          className="bx--grid-head__resize-marker"
          style={{
            left: `${this.state.leftMarkerPosition}px`,
            top: `${this.state.markerPositionTop}px`,
          }}
        />
        <div
          className="bx--grid-head__resize-marker"
          style={{
            left: `${this.state.rightMarkerPosition}px`,
            top: `${this.state.markerPositionTop}px`,
          }}
        />
        <div className="bx--grid-head__row">
          {cells}
        </div>
      </div>
    );
  }
}

GridHead.defaultPropTypes = {
  onSort: () => {},
  sort: {},
  updateColumns: () => {},
};

GridHead.propTypes = {
  columns: PropTypes.array.isRequired,
  onSort: PropTypes.func,
  sort: PropTypes.shape({
    direction: PropTypes.string,
    property: PropTypes.string,
  }),
  updateColumns: PropTypes.func,
};
