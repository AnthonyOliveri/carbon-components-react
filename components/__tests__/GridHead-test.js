import GridHead from '../GridHead';
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';

jest.unmock('../GridHead');

const shallowRender = props => shallow(<GridHead {...props} />);
const fullRender = props => mount(<GridHead {...props} />, { attach: true });

describe('Grid/Head', () => {
  describe('check if props are rendered correctly', () => {
    const columns = [
      {
        name: 'column1',
        title: 'column1',
        render: data => data.column1,
        width: 50,
      },
      {
        name: 'column2',
        title: 'column2',
        sortable: true,
        render: data => data.column2,
        width: 25,
      },
      {
        name: 'column3',
        title: 'column3',
        render: data => data.column3,
        width: 25,
      },
    ];
    describe('standard header - ASC sort', () => {
      const wrapper = shallowRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: jest.fn(),
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('standard header - DESC sort', () => {
      const wrapper = shallowRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'DESC',
          property: 'column2',
        },
        updateColumns: jest.fn(),
      });

      it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('onSort is called correctly - if sortable is true', () => {
    const columns = [
      {
        name: 'column1',
        title: 'column1',
        render: data => data.column1,
        width: 50,
      },
      {
        name: 'column2',
        title: 'column2',
        sortable: true,
        render: data => data.column2,
        width: 25,
      },
      {
        name: 'column3',
        title: 'column3',
        render: data => data.column3,
        width: 25,
      },
    ];
    const onSortMock = jest.fn();
    const wrapper = fullRender({
      columns,
      onSort: onSortMock,
      sort: {
        direction: 'ASC',
        property: 'column2',
      },
      updateColumns: jest.fn(),
    });

    it('onSort is triggered on click', () => {
      wrapper.find('.bx--grid-head__cell').at(1).simulate('click');
      expect(onSortMock).toBeCalledWith(columns[1]);
    });
  });

  describe('check if dragStart is performed correctly', () => {
    const columns = [
      {
        name: 'column1',
        title: 'column1',
        render: data => data.column1,
        width: 50,
      },
      {
        name: 'column2',
        title: 'column2',
        sortable: true,
        render: data => data.column2,
        width: 25,
      },
      {
        name: 'column3',
        title: 'column3',
        render: data => data.column3,
        width: 25,
      },
    ];
    const wrapper = fullRender({
      columns,
      onSort: jest.fn(),
      sort: {
        direction: 'ASC',
        property: 'column2',
      },
      updateColumns: jest.fn(),
    });

    ReactDOM.findDOMNode(
      wrapper.node.cells.column1
    ).getBoundingClientRect = jest.fn(() => ({
      width: 120,
      height: 120,
      top: 200,
      left: 100,
      bottom: 0,
      right: 0,
    }));
    wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragStart', {
      clientX: 220,
    });

    it('state is updated correctly on dragStart', () => {
      expect(wrapper.state()).toEqual({
        columns,
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 220,
        },
        leftMarkerPosition: 100,
        rightMarkerPosition: 220,
        markerPositionTop: 200,
      });
    });
  });

  describe('check if dragOver is performed correctly', () => {
    const columns = [
      {
        name: 'column1',
        title: 'column1',
        render: data => data.column1,
        width: 50,
      },
      {
        name: 'column2',
        title: 'column2',
        sortable: true,
        render: data => data.column2,
        width: 25,
      },
      {
        name: 'column3',
        title: 'column3',
        render: data => data.column3,
        width: 25,
      },
    ];
    describe('markerPosition gives more than 5% of grid size', () => {
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: jest.fn(),
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 220,
        },
        leftMarkerPosition: 100,
        rightMarkerPosition: 220,
        markerPositionTop: 200,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 0,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragOver', {
        clientX: 200,
      });

      it('rightMarkerPosition is updated', () => {
        expect(wrapper.state()).toEqual({
          columns,
          resizeColumn: {
            column: columns[0],
            dragStartLocation: 220,
          },
          leftMarkerPosition: 100,
          rightMarkerPosition: 200,
          markerPositionTop: 200,
        });
      });
    });

    describe('markerPosition gives less than 5% of grid size', () => {
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: jest.fn(),
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 220,
        },
        leftMarkerPosition: 100,
        rightMarkerPosition: 220,
        markerPositionTop: 200,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 0,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragOver', {
        clientX: 110,
      });

      it('rightMarkerPosition is unchanged', () => {
        expect(wrapper.state()).toEqual({
          columns,
          resizeColumn: {
            column: columns[0],
            dragStartLocation: 220,
          },
          leftMarkerPosition: 100,
          rightMarkerPosition: 220,
          markerPositionTop: 200,
        });
      });
    });
  });

  describe('check if dragEnd is performed correctly', () => {
    describe('column 1 is reduced to half', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 25,
        },
      ];
      const updateColumnsMock = jest.fn();
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: updateColumnsMock,
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 250,
        },
        leftMarkerPosition: 0,
        rightMarkerPosition: 250,
        markerPositionTop: 0,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column1
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 250,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column2
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 250,
        bottom: 0,
        right: 125,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column3
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 375,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragEnd', {
        clientX: 125,
      });

      const recalculatedColumns = [
        {
          name: 'column1',
          title: 'column1',
          width: 25,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 37.5,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 37.5,
        },
      ];
      it('state is updated correctly on dragEnd', () => {
        expect(wrapper.state()).toEqual({
          columns: recalculatedColumns,
          resizeColumn: null,
          leftMarkerPosition: -9999,
          rightMarkerPosition: -9999,
          markerPositionTop: -9999,
        });
      });

      it('props.updateColumns is called with the updated columns', () => {
        expect(updateColumnsMock).toBeCalledWith(recalculatedColumns);
      });
    });

    describe('column 2 is reduced to 80%', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 25,
        },
      ];
      const updateColumnsMock = jest.fn();
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: updateColumnsMock,
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[1],
          dragStartLocation: 375,
        },
        leftMarkerPosition: 0,
        rightMarkerPosition: 375,
        markerPositionTop: 0,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column1
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 250,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column2
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 250,
        bottom: 0,
        right: 125,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column3
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 375,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(1).simulate('dragEnd', {
        clientX: 350,
      });

      const recalculatedColumns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 20,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 30,
        },
      ];
      it('state is updated correctly on dragEnd', () => {
        expect(wrapper.state()).toEqual({
          columns: recalculatedColumns,
          resizeColumn: null,
          leftMarkerPosition: -9999,
          rightMarkerPosition: -9999,
          markerPositionTop: -9999,
        });
      });

      it('props.updateColumns is called with the updated columns', () => {
        expect(updateColumnsMock).toBeCalledWith(recalculatedColumns);
      });
    });

    describe('column 1 is reduced to 4%', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 25,
        },
      ];
      const updateColumnsMock = jest.fn();
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: updateColumnsMock,
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 250,
        },
        leftMarkerPosition: 0,
        rightMarkerPosition: 250,
        markerPositionTop: 0,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column1
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 250,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column2
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 250,
        bottom: 0,
        right: 125,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column3
      ).getBoundingClientRect = jest.fn(() => ({
        width: 125,
        height: 100,
        top: 0,
        left: 375,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragEnd', {
        clientX: 15,
      });

      const recalculatedColumns = [
        {
          name: 'column1',
          title: 'column1',
          width: 5,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 47.5,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 47.5,
        },
      ];
      it('state is updated correctly on dragEnd', () => {
        expect(wrapper.state()).toEqual({
          columns: recalculatedColumns,
          resizeColumn: null,
          leftMarkerPosition: -9999,
          rightMarkerPosition: -9999,
          markerPositionTop: -9999,
        });
      });

      it('props.updateColumns is called with the updated columns', () => {
        expect(updateColumnsMock).toBeCalledWith(recalculatedColumns);
      });
    });

    describe('column 1 is increased to 150%', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 25,
        },
      ];
      const updateColumnsMock = jest.fn();
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: updateColumnsMock,
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[0],
          dragStartLocation: 500,
        },
        leftMarkerPosition: 0,
        rightMarkerPosition: 500,
        markerPositionTop: 0,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 1000,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column1
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 500,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column2
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 500,
        bottom: 0,
        right: 250,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column3
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 750,
        bottom: 0,
        right: 0,
      }));
      wrapper.find('.bx--grid-head-cell__handle').at(0).simulate('dragEnd', {
        clientX: 625,
      });

      const recalculatedColumns = [
        {
          name: 'column1',
          title: 'column1',
          width: 62.5,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 18.75,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 18.75,
        },
      ];
      it('state is updated correctly on dragEnd', () => {
        expect(wrapper.state()).toEqual({
          columns: recalculatedColumns,
          resizeColumn: null,
          leftMarkerPosition: -9999,
          rightMarkerPosition: -9999,
          markerPositionTop: -9999,
        });
      });

      it('props.updateColumns is called with the updated columns', () => {
        expect(updateColumnsMock).toBeCalledWith(recalculatedColumns);
      });
    });

    describe('column 2 is increased to 150%', () => {
      const columns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 25,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 25,
        },
      ];
      const updateColumnsMock = jest.fn();
      const wrapper = fullRender({
        columns,
        onSort: jest.fn(),
        sort: {
          direction: 'ASC',
          property: 'column2',
        },
        updateColumns: updateColumnsMock,
      });

      wrapper.setState({
        resizeColumn: {
          column: columns[1],
          dragStartLocation: 750,
        },
        leftMarkerPosition: 0,
        rightMarkerPosition: 750,
        markerPositionTop: 0,
      });
      ReactDOM.findDOMNode(
        wrapper.node
      ).getBoundingClientRect = jest.fn(() => ({
        width: 1000,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column1
      ).getBoundingClientRect = jest.fn(() => ({
        width: 500,
        height: 100,
        top: 0,
        left: 0,
        bottom: 0,
        right: 500,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column2
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 500,
        bottom: 0,
        right: 250,
      }));
      ReactDOM.findDOMNode(
        wrapper.node.cells.column3
      ).getBoundingClientRect = jest.fn(() => ({
        width: 250,
        height: 100,
        top: 0,
        left: 750,
        bottom: 0,
        right: 0,
      }));

      wrapper.find('.bx--grid-head-cell__handle').at(1).simulate('dragEnd', {
        clientX: 875,
      });

      const recalculatedColumns = [
        {
          name: 'column1',
          title: 'column1',
          width: 50,
        },
        {
          name: 'column2',
          title: 'column2',
          sortable: true,
          width: 37.5,
        },
        {
          name: 'column3',
          title: 'column3',
          width: 12.5,
        },
      ];
      it('state is updated correctly on dragEnd', () => {
        expect(wrapper.state()).toEqual({
          columns: recalculatedColumns,
          resizeColumn: null,
          leftMarkerPosition: -9999,
          rightMarkerPosition: -9999,
          markerPositionTop: -9999,
        });
      });

      it('props.updateColumns is called with the updated columns', () => {
        expect(updateColumnsMock).toBeCalledWith(recalculatedColumns);
      });
    });
  });
});
