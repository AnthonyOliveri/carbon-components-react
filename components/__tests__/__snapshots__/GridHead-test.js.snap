exports[`Grid/Head check if props are rendered correctly standard header - ASC sort renders correctly 1`] = `
<div
  className="bx--grid-head">
  <div
    className="bx--grid-head__resize-marker"
    style={
      Object {
        "left": "-9999px",
        "top": "-9999px",
      }
    } />
  <div
    className="bx--grid-head__resize-marker"
    style={
      Object {
        "left": "-9999px",
        "top": "-9999px",
      }
    } />
  <div
    className="bx--grid-head__row">
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "ASC",
          "property": "column2",
        }
      } />
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "ASC",
          "property": "column2",
        }
      } />
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "ASC",
          "property": "column2",
        }
      } />
  </div>
</div>
`;

exports[`Grid/Head check if props are rendered correctly standard header - DESC sort renders correctly 1`] = `
<div
  className="bx--grid-head">
  <div
    className="bx--grid-head__resize-marker"
    style={
      Object {
        "left": "-9999px",
        "top": "-9999px",
      }
    } />
  <div
    className="bx--grid-head__resize-marker"
    style={
      Object {
        "left": "-9999px",
        "top": "-9999px",
      }
    } />
  <div
    className="bx--grid-head__row">
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column1",
          "render": [Function],
          "title": "column1",
          "width": 50,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "DESC",
          "property": "column2",
        }
      } />
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column2",
          "render": [Function],
          "sortable": true,
          "title": "column2",
          "width": 25,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "DESC",
          "property": "column2",
        }
      } />
    <GridHeadCell
      className="bx--grid-head__cell"
      column={
        Object {
          "name": "column3",
          "render": [Function],
          "title": "column3",
          "width": 25,
        }
      }
      dragEnd={[Function]}
      dragOver={[Function]}
      dragStart={[Function]}
      onSort={[Function]}
      sort={
        Object {
          "direction": "DESC",
          "property": "column2",
        }
      } />
  </div>
</div>
`;
